examples/Phonon_Caustics/src/Caustic_PhononDetectorConstruction.cc

```diff
@@ -0,0 +1,199 @@
+
+
+//
+//20240110 Israel Hernandez -- Illinois Institute of Technology
+
+
+#include "Caustic_PhononDetectorConstruction.hh"
+#include "Caustic_PhononSensitivity.hh"
+#include "G4CMPLogicalBorderSurface.hh"
+#include "G4CMPPhononElectrode.hh"
+#include "G4CMPSurfaceProperty.hh"
+#include "G4Box.hh"
+#include "G4Colour.hh"
+#include "G4GeometryManager.hh"
+#include "G4LatticeLogical.hh"
+#include "G4LatticeManager.hh"
+#include "G4LatticePhysical.hh"
+#include "G4LogicalVolume.hh"
+#include "G4LogicalVolumeStore.hh"
+#include "G4Material.hh"
+#include "G4MaterialPropertiesTable.hh"
+#include "G4NistManager.hh"
+#include "G4PVPlacement.hh"
+#include "G4PhysicalVolumeStore.hh"
+#include "G4RunManager.hh"
+#include "G4SDManager.hh"
+#include "G4SolidStore.hh"
+#include "G4SystemOfUnits.hh"
+#include "G4TransportationManager.hh"
+#include "G4Tubs.hh"
+#include "G4UserLimits.hh"
+#include "G4VisAttributes.hh"
+#include "Caustic_PhononConfigManager.hh"
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo....
+
+Caustic_PhononDetectorConstruction::Caustic_PhononDetectorConstruction()
+  : fLiquidHelium(0),fBolometer(0),fOxigen(0),fSubstrate(0),
+    fWorldPhys(0), topSurfProp(0), wallSurfProp(0),
+    electrodeSensitivity(0), fConstructed(false) {;}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo....
+
+Caustic_PhononDetectorConstruction::~Caustic_PhononDetectorConstruction() {
+  delete topSurfProp;
+
+  delete wallSurfProp;
+}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo....
+
+G4VPhysicalVolume* Caustic_PhononDetectorConstruction::Construct()
+{
+  if (fConstructed) {
+    if (!G4RunManager::IfGeometryHasBeenDestroyed()) {
+      // Run manager hasn't cleaned volume stores. This code shouldn't execute
+      G4GeometryManager::GetInstance()->OpenGeometry();
+      G4PhysicalVolumeStore::GetInstance()->Clean();
+      G4LogicalVolumeStore::GetInstance()->Clean();
+      G4SolidStore::GetInstance()->Clean();
+    }
+    // Have to completely remove all lattices to avoid warning on reconstruction
+    G4LatticeManager::GetLatticeManager()->Reset();
+    // Clear all LogicalSurfaces
+    // NOTE: No need to redefine the G4CMPSurfaceProperties
+    G4CMPLogicalBorderSurface::CleanSurfaceTable();
+  }
+
+  Caustic_DefineMaterials();
+  Caustic_SetupGeometry();
+  fConstructed = true;
+
+  return fWorldPhys;
+}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo....
+
+void Caustic_PhononDetectorConstruction::Caustic_DefineMaterials()
+{
+  G4NistManager* nistManager = G4NistManager::Instance();
+
+  fLiquidHelium = nistManager->FindOrBuildMaterial("G4_AIR"); // to be corrected
+  fBolometer = nistManager->FindOrBuildMaterial("G4_Al");
+
+//This lines works for Sapphire
+//
+ fOxigen = nistManager->FindOrBuildMaterial("G4_O");
+fSubstrate = new G4Material("fSubstrate", 3.98*g/cm3, 2);
```

Comment:
Please don't use the name of your variable as the name of the material.  You're defining sapphire, so give it the name "Sapphire".  Then when some code prints out the material's `GetName()`, the output will be understandable.